### PR TITLE
added convenience overloads for code blocks in control flow

### LINKED ISSUE #751: Add Named / CodeBlock overload to control flow
I really like the `CodeBlock::addNamed` / `MethodSpec.Builder::addNamedCode` features as they make the generating code a lot more readable.

Unfortunately, when trying to work exclusively with named code, it becomes somewhat annoying to deal with the statement and control flow characters. It'd be great if you could add some convenience methods to `MethodSpec.Builder`

* `addNamedStatement`
* `beginNamedControlFlow`
* `nextNamedControlFlow`

Additionally, it'd be great if `beginControlFlow` and `nextControlFlow` could have overloads for accepting a `CodeBlock` parameter like the `addStatement` method. That would make it easier to work with helper methods, e.g.,

`method.beginControlFlow(CodeBlock.builder().addNamed("if ($field:N != null)", map).build());`

### PR DESCRIPTION
**Pull request by ennerf**
Convenience overloads to use CodeBlocks in control flows. This useful when working with named code and some special cases.

Fixes #751

### CODE REVIEW

**Comment by ennerf** [comment]
Fixed. Sorry about the oversight.

File: src/test/java/com/squareup/javapoet/MethodSpecTest.java [hunk 1]
```diff
@@ -416,4 +417,47 @@ static void staticMethod() {
...
         + "}\n");
   }
+
+  @Test public void controlFlowWithNamedCodeBlocks() {
+    Map<String, Object> m = new HashMap<>();
+    m.put("field", "valueField");
+    m.put("threshold", "5");
+
+    MethodSpec methodSpec = MethodSpec.methodBuilder("method")
+            .beginControlFlow(named("if ($field:N > $threshold:L)", m))
```
Comment (Egorand) [inline src/test/java/com/squareup/javapoet/MethodSpecTest.java]: Please change continuation indents to 4 spaces to be consistent with the rest of the codebase.

Comment (Egorand) [reply src/test/java/com/squareup/javapoet/MethodSpecTest.java]: Otherwise tests look good!

Comment (ennerf) [reply src/test/java/com/squareup/javapoet/MethodSpecTest.java]: Done. For the record, that file mixes continuation indents of 2, 4, and 8 spaces.

Comment (Egorand) [reply src/test/java/com/squareup/javapoet/MethodSpecTest.java]: Thanks! Yeah, should be fixed. 4 spaces is correct.

**Pull request merged**